lib/spack/spack/platforms/cray.py

```diff
@@ -104,9 +98,15 @@ def setup_platform_environment(cls, pkg, env):
         env.append_path("PKG_CONFIG_PATH", "/usr/lib64/pkgconfig")
         env.append_path("PKG_CONFIG_PATH", "/usr/local/lib64/pkgconfig")
 
+        if os.environ.get('CRAY_LD_LIBRARY_PATH'):
+            env.prepend_path('LD_LIBRARY_PATH',
+                             os.environ['CRAY_LD_LIBRARY_PATH'])
+
     @classmethod
     def detect(cls):
-        return os.environ.get('CRAYPE_VERSION') is not None
+        # If the cray module-based programming environment is available in
+        # MODULEPATH, then this is a cray platform.
+        return 'opt/cray' in os.environ.get('MODULEPATH', '')
```

Comment:
add a longer docstring for detect that describes why we do it this way, and the distinction we're drawing between module- and non-module cray environments.   Basically explain that if a cray machine doesn't have this , it's linux.